src/simulator/engine/d3-simulator-engine.ts

```diff
@@ -497,34 +557,152 @@ export class D3SimulatorEngine extends Emitter<D3SimulatorEvents> {
     }
   }
 
-  releaseNodes(nodes?: ISimulationNode[]) {
+  /**
+   * Releases specified nodes.
+   * If no nodes are provided, this function releases all nodes.
+   *
+   * @param {ISimulationNode[]} nodes Nodes that are going to be released. If undefined, all nodes get released.
+   */
+  unfixNodes(nodes?: ISimulationNode[]) {
     if (!nodes) {
       nodes = this._nodes;
     }
 
     for (let i = 0; i < nodes.length; i++) {
-      releaseNode(this._nodes[i]);
+      this.unfixNode(this._nodes[i]);
     }
   }
-}
 
-const fixNode = (node: ISimulationNode) => {
-  // fx and fy fix the node position in the D3 simulation.
-  node.fx = node.x;
-  node.fy = node.y;
-};
+  /**
+   * Fixes a node by setting its `fx` and `fy` properties to `x` and `y`.
+   * This function is called when disabling physics.
+   *
+   * @param {ISimulationNode} node Simulation node that is going to be fixed
+   */
+  private fixNode(node: ISimulationNode) {
+    if (node.sx === null || node.sx === undefined) {
+      node.fx = node.x;
+    }
+    if (node.sy === null || node.sy === undefined) {
+      node.fy = node.y;
+    }
+  }
 
-const releaseNode = (node: ISimulationNode) => {
-  node.fx = null;
-  node.fy = null;
-};
+  /**
+   * Releases a node if it's not sticky by setting its `fx` and `fy` properties to `null`.
+   * This function is called when enabling physics but the sticky property overpowers physics.
+   *
+   * @param {ISimulationNode} node Simulation node that is going to be released
+   */
+  private unfixNode(node: ISimulationNode) {
+    if (node.sx === null || node.sx === undefined) {
+      node.fx = null;
+    }
+    if (node.sy === null || node.sy === undefined) {
+      node.fy = null;
+    }
+  }
+
+  /**
+   * Sticks the specified nodes into place.
+   * This overpowers any physics state and also sticks the node to their current positions.
+   * If no nodes are provided, this function sticks all nodes.
+   *
+   * @param {ISimulationNode[]} nodes Nodes that are going to become sticky. If undefined, all nodes get sticked.
+   */
+  stickNodes(nodes?: ISimulationNode[]) {
+    if (!nodes) {
+      nodes = this._nodes;
+    }
+
+    for (let i = 0; i < nodes.length; i++) {
+      this.stickNode(this._nodes[i]);
+    }
+  }
+
+  /**
+   * Sticks all nodes thath have a defined position (x and y coordinates).
+   * This function should be called when the user initially sets up or merges some data.
+   * If the user provided nodes already have defined `x` **or** `y` properties, they are treated as _"sticky"_.
+   * Only the specified axis gets immobilized.
+   *
+   * @param {ID3SimulatorGraph} data Graph data.
+   * @return {ID3SimulatorGraph} Graph data with attached `{fx, sx}`, and/or `{fy, sy}` coordinates.
+   */
+  private _fixDefinedNodes(data: ID3SimulatorGraph): ID3SimulatorGraph {
+    // TODO(dlozic): Question: should this function be extracted or should i use `this.data` everywhere and remove inputs/outputs?
```

Comment:
Well, here is how I see it: I am bit confused with the name, it says `fixDefinedNodes`. Ok, definition is, I assume if `x` or `y` is defined. Then it sets `fx` and `sx`. When I check the `fixNodes` (`fixNode`) method, it is not working like that, it is just setting `fx`. So, to recap, there is some inconsistency here - naming-wise.

Reply:
Regarding the `data` input, as far as I can see it is used correctly (where `fixDefinedNodes` is used), meaning `data` is valid as an input here.

Reply:
> 

Yeah you're right. Should be `_stickDefinedNodes`.
Regarding naming, do the other methods make sense? (`fix` <-> `unfix`, `stick` <-> `unstick`)?

Reply:
> Regarding the `data` input, as far as I can see it is used correctly (where `fixDefinedNodes` is used), meaning `data` is valid as an input here.

But should the method here go outside the class? Should it be a function and not a class method?

Reply:
<rename and remove (release)>